SignalServiceKit/src/Util/NSUserDefaults+OWS.m

```diff
@@ -0,0 +1,38 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "NSUserDefaults+OWS.h"
+#import "TSConstants.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation NSUserDefaults (OWS)
+
++ (NSUserDefaults *)appUserDefaults
+{
+    return [[NSUserDefaults alloc] initWithSuiteName:SignalApplicationGroup];
+}
+
++ (void)migrateToSharedUserDefaults
+{
+    NSUserDefaults *appUserDefaults = self.appUserDefaults;
+
+    NSDictionary<NSString *, id> *dictionary = [NSUserDefaults standardUserDefaults].dictionaryRepresentation;
+    for (NSString *key in dictionary) {
+        id value = dictionary[key];
+        OWSAssert(value);
+        [appUserDefaults setObject:value forKey:key];
+    }
```

Comment:
Do we want to delete the old preferences at this point?

Reply:
Great question. I decided to leave them in place for a few reasons.

The main thing is that we've never stored much or anything that important in `NSUserDefaults`.